tests/unit/test_coresubset.py

```diff
@@ -1426,6 +1429,361 @@ def test_rp_cholesky_fit_invalid_data(self):
+        )
+        random_coreset = coresubset_object_random.coreset
+
+        # Define a metric and compare quality of produced coresets
+        metric = coreax.metrics.MMD(kernel=kernel)
+        not_random_metric = metric.compute(x, fitted_coresubset)
+        random_metric = metric.compute(x, random_coreset)
+        self.assertLess(float(not_random_metric), float(random_metric))
+
+        # Create a coreset via SteinThinning with refinement, to check refinement
+        # improves the coreset quality
+        coresubset_object_not_random.refine()
+        refined_fitted_coresubset = coresubset_object_not_random.coreset
+
+        # Compare quality of refined coreset to non-refined coreset
+        refined_not_random_metric = metric.compute(x, refined_fitted_coresubset)
+        self.assertLess(float(refined_not_random_metric), float(not_random_metric))
+
+    def test_fit_comparison_to_random_and_refined_not_unique(self) -> None:
+        """
+        Test the non-unique fit method of the SteinThinning class with a simple example.
+
+        The test checks that a coreset generated via SteinThinning has an improved
+        quality (measured by maximum mean discrepancy) than one generated by random
+        sampling. We further check that if the coreset generated by SteinThinning is
+        refined, the quality improves yet again.
+        """
+        # Define specific test instance setup
+        # kernel = coreax.kernel.SquaredExponentialKernel()
+        num_data_points = 100
+
+        # Define some data - sufficiently large that we would not expect a random sample
+        # to typically compete with a SteinThinning approach
+        generator = np.random.default_rng(self.random_data_generation_key)
+        x = generator.random((num_data_points, self.dimension))
+        data = coreax.data.ArrayData.load(x)
+        num_samples_length_scale = min(num_data_points, 1_000)
+        idx = generator.choice(num_data_points, num_samples_length_scale, replace=False)
+        length_scale = float(coreax.kernel.median_heuristic(x[idx]))
+        # Define specific test instance setup
+        kernel = coreax.kernel.PCIMQKernel(length_scale=length_scale)
+
+        # Create a SteinThinning object
+        coresubset_object_not_random = coreax.coresubset.SteinThinning(
+            self.random_key,
+            kernel=kernel,
+            refine_method=coreax.refine.RefineRegular(),
+            unique=False,
+        )
+
+        # Apply SteinThinning on the dataset, and record the coreset for comparison
+        coresubset_object_not_random.fit(
+            original_data=data, strategy=coreax.reduction.SizeReduce(self.coreset_size)
+        )
+        fitted_coresubset = coresubset_object_not_random.coreset
+
+        # Create a random refinement object and generate a coreset, for comparison
+        coresubset_object_random = coreax.coresubset.RandomSample(
+            random_key=self.random_key, unique=True
```

Comment:
Should random be `unique=True` in this test, or should it match `SteinThining` and have `unique=False`?
```suggestion
            random_key=self.random_key, unique=False
```

Reply:
Done.